server/graphql/programs/src/mutations/patient/update.rs

```diff
@@ -1,21 +1,29 @@
 use async_graphql::*;
+use chrono::NaiveDate;
 use graphql_core::{
     standard_graphql_error::{validate_auth, StandardGraphqlError},
     ContextExt,
 };
+use graphql_general::GenderInput;
 use graphql_types::types::patient::PatientNode;
 use service::{
     auth::{Resource, ResourceAccessRequest},
-    programs::patient::{UpdatePatient, UpdatePatientError},
+    programs::patient::{UpdateNamePatient, UpdateNamePatientError},
 };
+use util::inline_init;
 
+/// All fields in the input object will be used to update the patient record.
+/// This means that the caller also has to provide the fields that are not going to change.
+/// For example, if the last_name is not provided the last_name in the patient record will be clear.
 #[derive(InputObject)]
 pub struct UpdatePatientInput {
```

Comment:
I choose this behaviour for all the fields. Had look and it seems in other place we have a mix of selectively updating field if set, or always updating a field...

Reply:
I was mainly thinking about the use-case where you want to unset something, i.e. not updating VS setting to null